### DISCUSSION TITLE
mergeAndConcat

### DISCUSSION THREAD

**Discussion by silveltman** [remeda/remeda#953] [Suggest a Function]
similar to mergeDeep but concatting arrays with the same key, instead of overwriting. 

This one is awesome, with good ts support: https://github.com/mesqueeb/merge-anything

edit: I think mergeAndConcat (or mergeConcat) would be a better name, rather then mergeDeepConcat for example. Reason being that in order to merge AND concat the value already is sure to be a combination of object and array, therefore must be deeper then one layer

**Comment by cjquines** [comment]
my proposed api here would be to reuse mergeDeep with an options param, like [deno's deepMerge](https://deno.land/std@0.224.0/collections/deep_merge.ts?s=deepMerge)

**Comment by terinjokes** [comment]
The ramda function I was looking at migrating from is `R.mergeDeepWith` where `R.concat` is provided as the function. That said, I have no examples in my projects of `R.mergeDeepWith` with any other function than `R.concat`, but wouldn't rule it out as useful to have.
